### ISSUE TITLE
Questions about the detection results of Focalformer

### ISSUE THREAD

**Issue by ShXiuWei1025** [NVlabs/FocalFormer3D#9]
Hello. @chenyilun95

I would like to ask if you still have the detection result file (.json) of the full nuscenes dataset (800 scenes, including 700 trainval and 150 val)? I want to test some tracking methods on it, because my current computer is very slow in processing the detection.

Many thanks.

**Comment by ShXiuWei1025** [comment]
only the Lidar detection

**Comment by chenyilun95** [comment]
Thank you for your inquiry! We have provided the testing set results from the model trained on trainval sets in README file. The results for the train+val set can be obtained from the model trained on train set only or trainval set. To clarify, which set of results would you like to obtain?

**Comment by ShXiuWei1025** [comment]
I want to obtain the detection results of train set, and if possible, the result from trainval set. 

I have a problem for the Nuscenes lidar data info generation via mmdetection3d. Why it always requires the camera data? I only want to do the lidar data detection. I use the command:  python tools/create_data.py nuscenes --root-path ./data/nuscenes --out-dir ./data/nuscenes --extra-tag nuscenes.

**Comment by ShXiuWei1025** [comment]
I mean if I only want to work on Lidar detection. Then what should I do? It seems this project only contains multi-model processing. I download the pre-trained model "FocalFormer3D_L.pth", while in this code:
`data_root = 'data/nuscenes/'
input_modality = dict(
    use_lidar=True,
    use_camera=True,
    use_radar=False,
    use_map=False,
    use_external=False)`

why "use_camera=True"?

**Comment by ShXiuWei1025** [comment]
Sorry for so many questions. It's my first using mmdetection3d, so I have a lot issues when using.

**Comment by chenyilun95** [comment]
1. I mean the results you want is generated from the model trained on train set only or the model trained on trainval set?
2. I conducted both fusion and lidar experiments. You might need to check if the lidar-only option in create_data for mmdet3d. Anyway, you can try modifying the code in the dataset generation if you don't want camera.

**Comment by chenyilun95** [comment]
> I mean if I only want to work on Lidar detection. Then what should I do? It seems this project only contains multi-model processing. I download the pre-trained model "FocalFormer3D_L.pth", while in this code: `data_root = 'data/nuscenes/' input_modality = dict( use_lidar=True, use_camera=True, use_radar=False, use_map=False, use_external=False)`
> 
> why "use_camera=True"?

It's the input setting. But it's not used in the model. You can see the following config "input_img=False" and the image backbone is commented. You can debug the intermediate model input/output if you are able to run the code.

**Comment by ShXiuWei1025** [comment]
Thanks for your reply. I want the results generated from the model trained on train set only

**Comment by chenyilun95** [comment]
It's available at [link](https://mycuhk-my.sharepoint.com/:u:/g/personal/1155113995_link_cuhk_edu_hk/EU1qCau3Rk1LtGE1Zlgv7AoB4gIHUM_LVcB2-o6WsYTSfw?e=dN1eTo). Note this is generated from the model using train only dataset. If you get better results, you can also directly use the json result in the README, which is trained using the trainval set😄.

**Comment by ShXiuWei1025** [comment]
Thanks for providing the result. Could I also request for these detection results (using trainval model)? I want to train some tracking methods through focalformer's head and detection results. I come to realize that the test set detection result (in the README) is using trainval set model and it would be better to match these two models. 

Thanks again for your help and patient.

**Comment by chenyilun95** [comment]
I checked the model weight files and I did not save the trainval checkpoints on my hardware. I may need to train it again, which will take some time.

**Comment by ShXiuWei1025** [comment]
i appreciate your help

**Comment by chenyilun95** [comment]
@ShXiuWei1025  I have uploaded the trainval results using the trainval model [here](https://mycuhk-my.sharepoint.com/:u:/g/personal/1155113995_link_cuhk_edu_hk/EW_Twp8wTa5GqP7zGtx64SgBKOzZej0IX8hmlw2hhXzWpA?e=f8x6ZD) :).

**Comment by ShXiuWei1025** [comment]
Thank you very much yilun. I really appreciate your help!

**Comment by mx2013713828** [comment]
@ShXiuWei1025 Hello, I noticed that you test some tracking methods , now I also start learning object tracking , can you share your experience about tracking methods?

**Comment by mx2013713828** [comment]
@ShXiuWei1025 I watch the nuscence detection result ,and it have 'class','translation' ,'size', 'rotation','velocity','score'. but now my detector is pointpillar and trained in kitti dataset ,its result is 'class,x,y,z,w,l,h,roll,pitch,yaw,score' ,how could I use it ?

**Comment by ShXiuWei1025** [comment]
> @ShXiuWei1025 I watch the nuscence detection result ,and it have 'class','translation' ,'size', 'rotation','velocity','score'. but now my detector is pointpillar and trained in kitti dataset ,its result is 'class,x,y,z,w,l,h,roll,pitch,yaw,score' ,how could I use it ?

I haven't used Kitti before. But in Nuscenes, the rotation only represents yaw, size is wlh, translation is xyz and velocity is v_x, v_y

**Comment by chenyilun95** [comment]
Hi, we use [SimpleTrack](https://github.com/tusen-ai/SimpleTrack) as the tracker in our experiment. The format of the KITTI dataset is different from that of nuScenes. I think one simple way is to set a constant value for v_x and v_y. I recommend that you inquire about it in that repo.

**Comment by ShXiuWei1025** [comment]
> @ShXiuWei1025 Hello, I noticed that you test some tracking methods , now I also start learning object tracking , can you share your experience about tracking methods?

Currently it is just a test version, I am still trying other strategies. But I believe the combination of neural networks and statistical methods (e.g. Bayesian framework) would be a good way in MOT.

**Comment by mx2013713828** [comment]
> Hi, we use [SimpleTrack](https://github.com/tusen-ai/SimpleTrack) as the tracker in our experiment. The format of the KITTI dataset is different from that of nuScenes. I think one simple way is to set a constant value for v_x and v_y. I recommend that you inquire about it in that repo.

Thanks for your reply~I will try ask SimpleTrack's author~

**Comment by mx2013713828** [comment]
> > @ShXiuWei1025 Hello, I noticed that you test some tracking methods , now I also start learning object tracking , can you share your experience about tracking methods?
> 
> Currently it is just a test version, I am still trying other strategies. But I believe the combination of neural networks and statistical methods (e.g. Bayesian framework) would be a good way in MOT.

ok,I got it. thanks for your reply~
